includes/controllers/class-message.php

```diff
@@ -67,6 +74,50 @@ public function __construct( $args = [] ) {
 		parent::__construct( $args );
 	}
 
+    /**
+     * Blocks user.
+     *
+     * @param WP_REST_Request $request API request.
+     * @return WP_Rest_Response
+     */
+    public function block_user( $request ) {
+
+        // Check authentication.
+        if ( ! is_user_logged_in() ) {
+            return hp\rest_error( 401 );
+        }
+
+        // Get blocked users.
+        $blocked_users = (array) get_user_meta( get_current_user_id(), hp\prefix( 'blocked_users' ), true );
```

Comment:
Maybe let's add a helper for this, this way we'll wrap this WordPress-level logic into a single function and will be able to customize it later in a single place. If I remember correctly the Favorites extension has something similar, it uses a helper